gui/cache.go

```diff
@@ -156,9 +161,14 @@ func (c *Cache) getMinedWorkByAccount(first, last int, accountID string) (int, [
 	}
 
 	count := len(allWork)
+
+	if last >= count+(last-first) {
```

Comment:
I don't understand the logic here. Last is clamped to be <= `count` so it doesnt need to be checked. I think this should just be...

```
if first > count {
	return 0, allWork[0:0], fmt.Errorf("Blocks by account request is out of range. Maximum %v, Requested %v", count, first)
}
```